pkg/util/k8s/pod.go

```diff
@@ -32,3 +45,57 @@ func GetPodContainerNames(pod *v1.Pod) []string {
 	}
 	return names
 }
+
+var (
+	netdefutilsSetNetworkStatus = netdefutils.SetNetworkStatus
+	netdefutilsGetNetworkStatus = netdefutils.GetNetworkStatus
+)
+
+func UpdatePodNetworkStatusAnnotation(kubeClient clientset.Interface, ctx context.Context, netStatus []netdefv1.NetworkStatus, podName, podNamespace string, isPrimary bool) error {
+	// Update the Pod's network status annotation
+	if netStatus == nil {
+		return nil
+	}
+	var podItem *v1.Pod
+	if err := wait.PollUntilContextTimeout(ctx, time.Second, 3*time.Second, true, func(ctx context.Context) (done bool, err error) {
+		podItem, err = kubeClient.CoreV1().Pods(podNamespace).Get(ctx, podName, metav1.GetOptions{})
+		if err != nil {
+			klog.V(2).InfoS("Get Pod error", "Pod", klog.KRef(podNamespace, podName), "error", err)
+			if errors.IsNotFound(err) {
+				return false, err
+			}
+			return false, nil
+		}
+		return true, nil
+	}); err != nil {
+		klog.ErrorS(err, "Failed to get Pod after retries", "Pod", klog.KRef(podNamespace, podName))
+		return err
```

Comment:
we try not to log an error and return an error, as this tends to lead to duplicate error logs. Do one or the other.